datasets/hk/legco/crawler.py

```diff
@@ -0,0 +1,186 @@
+"""
+Crawler for extracting names and constituencies of Hong Kong
+Legislative Council members.
+"""
+
+from typing import Dict, Union, NamedTuple, Any
+
+from zavod import Context
+from zavod import helpers as h
+from zavod.logic.pep import categorise
+
+
+MEMBER_URL_FORMAT = (
+    "https://app4.legco.gov.hk/mapi/{lang}/api/LASS/getMember?member_id={member_id}"
+)
+
+
+class MemberPages(NamedTuple):
+    """Member pages in English, Traditional and Simplified Chinese"""
+
+    en: Dict[str, Any]
+    hant: Dict[str, Any]
+    hans: Dict[str, Any]
+
+
+def crawl_member_pages(context: Context, member_id: int) -> MemberPages:
+    # Retrieve member pages in English, traditional and simplified Chinese
+    pages = []
+    for lang in "en", "tc", "cn":
+        member_url = MEMBER_URL_FORMAT.format(lang=lang, member_id=member_id)
+        response = context.fetch_json(member_url, cache_days=7)
+        pages.append(response["data"])
+    return MemberPages(*pages)
+
+
+def crawl_person(
+    context: Context,
+    member: Dict[str, Union[str, int]],
+    pages: MemberPages,
+):
+    """Fetch personal information for a Legislative Council member and
+    create a Person entity."""
+    person = context.make("Person")
+    person.id = context.make_id(member.pop("member_id"), member.pop("search_key"))
+    context.log.info("Unique ID {person_id}".format(person_id=person.id))
+    # Add names in English and both Chinese writing systems
+    h.apply_name(person, full=pages.en.pop("name"), lang="en")
+    h.apply_name(person, full=pages.hant.pop("name"), lang="zh_hant")
+    h.apply_name(person, full=pages.hans.pop("name"), lang="zh_hans")
+    for email in pages.en.pop("email_address", []):
+        if email:
+            context.log.info("Email: {email}".format(email=email))
```

Comment:
you can change these to debug-level once you're close to done

Reply:
done!